R/Archiver.R

```diff
@@ -0,0 +1,163 @@
+#' @title `Archiver`
+#' @keywords internal
+Archiver <- R6::R6Class( # nolint: object_name_linter.
+  classname = "Archiver",
+  public = list(
+    #' @description Returns a `Archiver` object.
+    #'
+    #' @return a `Archiver` object
+    #' @examples
+    #' archiver <- teal.reporter:::Archiver$new()
+    #'
+    initialize = function() {
+      invisible(self)
+    },
+    #' @description Finalizes an `Archiver` object.
+    finalize = function() {},
+    #' @description Pure virtual method for reading an `Archiver`.
+    read = function() {
+      # returns Reporter instance
+      stop("Pure virtual method.")
+    },
+    #' @description Pure virtual method for writing an `Archiver`.
+    write = function() {
+      stop("Pure virtual method.")
+    }
+  ),
+  lock_objects = TRUE,
+  lock_class = TRUE
+)
+
+#' @title `RDSArchiver`
+#' @keywords internal
+FileArchiver <- R6::R6Class( # nolint: object_name_linter.
+  classname = "RDSArchiver",
```

Comment:
Classname does not match the binding name.

Reply:
thanks